mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/RunningWorker.java

```diff
@@ -88,37 +88,21 @@ public RunningWorker(Builder builder) {
         this.stageTotalWorkersObservable = builder.stageTotalWorkersObservable;
         this.jobSchedulingInfoObservable = builder.jobSchedulingInfoObservable;
 
-        this.onTerminateCallback = new Action0() {
-            @Override
-            public void call() {
-                blockUntilTerminate.countDown();
-            }
-        };
-        this.onCompleteCallback = new Action0() {
-            @Override
-            public void call() {
-                logger.info("JobId: " + jobId + " stage: " + stageNum + ", completed");
-                // setup a timeout to call forced exit as sure way to exit
-                new Thread() {
-                    @Override
-                    public void run() {
-                        try {
-                            sleep(3000);
-                            System.exit(1);
-                        } catch (Exception e) {
-                            logger.error("Ignoring exception during exit: " + e.getMessage(), e);
-                        }
-                    }
-                }.start();
-                signalCompleted();
-            }
-        };
-        this.onErrorCallback = new Action1<Throwable>() {
-            @Override
-            public void call(Throwable t) {
-                signalFailed(t);
-            }
+        this.onTerminateCallback = blockUntilTerminate::countDown;
+        this.onCompleteCallback = () -> {
+            logger.info("JobId: " + jobId + " stage: " + stageNum + ", completed");
```

Comment:
Can we replace string concatenation with parameterized / placeholder in logging? It applies to other logging lines in this module.

```
logger.info("JobId: {} stage: {} completed", jobId, stageNum);
``` 
See this doc for advantages: https://www.tutorialspoint.com/slf4j/slf4j_parameterized_logging.htm